codegen/tests/trait_and_type_generic_param.rs

```diff
@@ -0,0 +1,28 @@
+use delegation::delegate;
+
+#[delegate]
+trait Named<N> {
+    fn name(&self) -> N;
+}
+
+struct User(String);
+impl Named<String> for User {
+    fn name(&self) -> String {
+        self.0.clone()
+    }
+}
+
+#[delegate(derive(
+    for<N> Named<N>
+    where
+        U: Named<N> + 'static;
+))]
+enum Case<U> {
```

Comment:
@50U10FCA7 using "either trait or type generics" is not enough, because doesn't support this case, while it should be supported. We should merge them instead.

Reply:
Additionally, let's return a hard error when `for<..>` syntax contains parameters named the same way as the type has already, to avoid accidental ambiguities.